app/src/main/java/com/github/steroidteam/todolist/view/ListSelectionActivity.java

```diff
@@ -15,30 +15,46 @@
 import androidx.constraintlayout.widget.ConstraintLayout;
 import com.firebase.ui.auth.AuthUI;
 import com.github.steroidteam.todolist.R;
+import com.github.steroidteam.todolist.database.Database;
+import com.github.steroidteam.todolist.database.DatabaseFactory;
 import com.github.steroidteam.todolist.model.todo.TodoList;
 import java.util.ArrayList;
 import java.util.List;
 
 public class ListSelectionActivity extends AppCompatActivity {
 
+    public static final String EXTRA_ID_TODO_LIST = "id_todo_list";
+
     private static todoListAdapter adapter;
     private ArrayList<TodoList> todoLists;
+    private Database database;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_list_selection);
 
         todoLists = new ArrayList<>();
-        TodoList todoList = new TodoList("Sweng project");
-        todoLists.add(todoList);
-        todoLists.add(new TodoList("Homework"));
-        todoLists.add(new TodoList("Some stuff"));
-        setTitle("TODO Lists");
 
+        setTitle("TODO Lists");
         ListView listView = findViewById(R.id.activity_list_selection_itemlist);
         adapter = new todoListAdapter(todoLists);
         setListViewSettings(listView);
+
+        database = DatabaseFactory.getDb();
+
+        database.getTodoListCollection()
+                .thenAccept(
+                        (todoListCollection -> {
+                            for (int i = 0; i < todoListCollection.getSize(); i++) {
```

Comment:
If we decide to stick with the `TodoListCollection` (instead of a plain `ArrayList`), it might be worth making it an `Iterable`, for situations like this one. See [this](https://stackoverflow.com/a/21512315) for reference.